# Microbiome Data Analysis using Phylogenetic Trees with a Bioinformatics Pipeline 
Understanding phylogenetic relationships between different species are crucial for evolutionary studies. Reconstructing the phylogenetic species tree, a branching diagram, is particularly useful in inferring evolutionary relationships. For example, the tree-of-life provides a remarkable view of organizing principles of the biological world. So, the exact species tree to be reconstructed is necessary, but the process of reconstructing the species or gene tree is very tedious.                   
 
Here, we developed an easy-to-use pipeline that conveniently and efficiently reconstructs species trees. this pipeline has many applications. We create a pipeline that will take in a list of taxa names, generated by previous metagenomic analysis of organisms, and develop a way to create a tree from them.       


# Pipeline Workflow Features      
   
*Inputs only include species names.       

*One python script to build newick file.     

*One python scirpt to visualize the tree.

*View trees using the phylo package.
